cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
@rishitesh Also it is not clear to me how a single StreamingContext obviates the need to make this thread-safe. Can't multiple concurrent job submissions use the same StreamingContext end up adding jars, then a later one actually starts the context?